### ISSUE TITLE
cli tests: failed version parsing with vendor suffix

### ISSUE THREAD

**Issue by maan2003** [fedimint/fedimint#6487]
semver version parsing is failing with vendor suffix is failing

https://github.com/fedimint/fedimint/blob/b3fcaa2fda1565004d9e377ed74d885b07b081f7/devimint/src/tests.rs#L979-L982

because it is not semver compatible

https://github.com/fedimint/fedimint/blob/b3fcaa2fda1565004d9e377ed74d885b07b081f7/fedimintd/src/fedimintd.rs#L266-L270

but even after fixing it, versions in both commands are different

```
bash-5.2$ fedimint-cli dev peer-version --peer-id 0
{
  "version": "0.4.4.fedi"
}
bash-5.2$ fedimintd --version
fedimintd 0.4.4 
```

**Comment by elsirion** [comment]
What do you think is the preferrable fix? Make the serialization format `x.y.z-vendor` or build a custom parser?

**Comment by maan2003** [comment]
vote for x.y.z-vendor

maybe we should just remove vendor suffix :shrug:

**Comment by elsirion** [comment]
Nah, knowing which arcane version someone is running is kinda useful. But dash syntax sounds good.

**Comment by dpc** [comment]
dash = "prerelease"
plus = "build"
